src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java

```diff
@@ -284,4 +304,9 @@ private MLMemorySpec getMLMemorySpec(Object mlMemory) {
 
         return builder.build();
     }
+
+    private Map<String, Object> getLLMFieldMap(String llmFieldMapString) {
+        BytesReference llmFieldBytes = new BytesArray(llmFieldMapString.getBytes(StandardCharsets.UTF_8));
+        return XContentHelper.convertToMap(llmFieldBytes, false, MediaTypeRegistry.JSON).v2();
```

Comment:
This method throws an `OpenSearchParseException`.   This has useful parsing information and a `status()` that can be used by the calling method.

While technically optional to add `throws` here, probably would help readability as we're doing the same as the method we're calling.